plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,259 @@
+              path_entry, source.separator, environment_variables, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY):
+          keys = set(source.keys)
+          for key_entry in keys:
+            if self._CheckKeyCompatibility(key_entry):
+              self.BuildFindSpecsFromRegistryArtifact(
+                key_entry, find_specs)
+        elif (source.type_indicator ==
+              artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_VALUE):
+          # TODO: Handle Registry Values Once Supported in dfwinreg.
+          logging.warning(('Unable to handle Registry Value, extracting '
+                           'key only: {0:s} ').format(source.key_value_pairs))
+
+          for key_pair in source.key_value_pairs:
+            keys = set()
+            keys.add(key_pair.get('key'))
+            for key_entry in keys:
+              if self._CheckKeyCompatibility(key_entry):
+                self.BuildFindSpecsFromRegistryArtifact(
+                  key_entry, find_specs)
+        else:
+          logging.warning(('Unable to handle artifact, plaso does not '
+                           'support: {0:s} ').format(source.type_indicator))
+
+    self._knowledge_base.SetValue(self.ARTIFACT_FILTERS, find_specs)
+
+  def BuildFindSpecsFromFileArtifact(
+      self, path_entry, separator, environment_variables, find_specs):
+    """Build find specifications from a FILE artifact type.
+
+    Args:
+      path_entry (str):  Current file system path to add.
+          environment variables.
+      separator (str): File system path separator.
+      environment_variables list(str):  Environment variable attributes used to
+          dynamically populate environment variables in key.
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for glob_path in self._ExpandRecursiveGlobs(path_entry):
+      for path in path_helper.PathHelper.ExpandUserHomeDirPath(
+          glob_path, self._knowledge_base.user_accounts):
+        if '%' in path:
+          path = path_helper.PathHelper.ExpandWindowsPath(
+              path, environment_variables)
+
+        if not path.startswith('/') and not path.startswith('\\'):
+          logging.warning((
+              'The path filter must be defined as an absolute path: '
+              '{0:s}').format(path))
+          continue
+
+        # Convert the path filters into a list of path segments and
+        # strip the root path segment.
+        path_segments = path.split(separator)
+
+        # If the source didn't specify a separator, '/' is returned by
+        # default from ForensicArtifacts, this is sometimes wrong.  Thus,
+        # need to check if '\' characters are still present and split on those.
+        if len(path_segments) == 1 and '\\' in path_segments[0]:
+          logging.warning('Potentially bad separator = {0:s} , trying \'\\\''
```

Comment:
please break line after `(`

Style guide is not clear about breaking before `.` but it seems to prefer implicit line break
https://google.github.io/styleguide/pyguide.html#Line_length

Seeing that elsewhere in the codebase breaking before `.` is not used I opt to break after `(` instead (for consistency sake).

Reply:
I opt to match the style of the rest of the codebase:

`bad separator = {0:s} , trying` => `bad separator: "{0:s}", trying`

Reply:
Done.